aspnetcore/security/blazor/server-side.md

```diff
@@ -0,0 +1,378 @@
+
+When a user visits a Blazor server-side site, the server creates a circuit in the server's memory. The circuit indicates to the browser what content to render and responds to events, such as when the user selects a button in the UI. To perform these actions, a circuit invokes JavaScript functions in the user's browser and .NET methods on the server. This two-way JavaScript-based interaction is referred to as [JavaScript interop (JS interop)](xref:blazor/javascript-interop).
+
+Because JS interop occurs over the Internet and the client uses a remote browser, Blazor server-side apps share most web app security concerns. This topic describes common threats to Blazor server-side apps and provides threat mitigation guidance focused on Internet-facing apps.
+
+In constrained environments, such as inside corporate networks or intranets, some of the mitigation guidance either:
+
+* Doesn't apply in the constrained environment.
+* Isn't worth the cost to implement because the security risk is low in a constrained environment.
+
+## Resource exhaustion
+
+Resource exhaustion can occur when a client interacts with the server and causes the server to consume excessive resources. Excessive resource consumption primarily affects:
+
+* [CPU](#cpu)
+* [Memory](#memory)
+* [Client connections](#client-connections)
+
+Denial of service (DoS) attacks usually seek to exhaust an app or server's resources. However, resource exhaustion isn't necessarily the result of an attack on the system. For example, finite resources can be exhausted due to high user demand. DoS is covered further in the [Denial of service (DoS) attacks](#denial-of-service-dos-attacks) section.
+
+Resources external to the Blazor framework, such as databases and file handles (used to read and write files), may also experience resource exhaustion. For more information, see <xref:performance/performance-best-practices>.
+
+### CPU
+
+CPU exhaustion can occur when one or more clients force the server to perform intensive CPU work.
+
+For example, consider a Blazor server-side app that calculates a *Fibonnacci number*. A Fibonnacci number is produced from a Fibonnacci sequence, where each number in the sequence is the sum of the two preceding numbers. The amount of work required to reach the answer depends on the length of the sequence and the size of the initial value. If the app doesn't place limits on a client's request, the CPU-intensive calculations may dominate the CPU's time and diminish the performance of other tasks.
+
+<!-- Should the preceding add something to the effect that CPU exhaustion can reach a point that it may even crash the server (or 'for all practical purposes crash the server because the server slows to a crawl')? -->
```

Comment:
I think the typical blurb is that we consider excessive resource consumption to be a security concern impacting **availability**. Where your server has crashed or not, if it's not responsive, that's the same outcome as a crash.